tests/benchmarks/peripheral_stress_test/src/flash_thread.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(flash, LOG_LEVEL_INF);
+
+#include <zephyr/drivers/flash.h>
+#include <zephyr/devicetree.h>
+#include "common.h"
+
+#if DT_HAS_COMPAT_STATUS_OKAY(jedec_spi_nor)
+#define FLASH_NODE DT_COMPAT_GET_ANY_STATUS_OKAY(jedec_spi_nor)
+#elif DT_HAS_COMPAT_STATUS_OKAY(nordic_qspi_nor)
+#define FLASH_NODE DT_COMPAT_GET_ANY_STATUS_OKAY(nordic_qspi_nor)
+#else
+#error Unsupported flash driver
+#define FLASH_NODE DT_INVALID_NODE
+#endif
+
+#define TEST_AREA_OFFSET	(0xff000)
+#define EXPECTED_SIZE		(512)
+
+static const struct device *const flash_dev = DEVICE_DT_GET(FLASH_NODE);
+
+/* Flash thread */
+static void flash_thread(void *arg1, void *arg2, void *arg3)
+{
+	ARG_UNUSED(arg1);
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+
+	int ret;
+	uint8_t buf[EXPECTED_SIZE];
+
+	atomic_inc(&started_threads);
+
+	if (!device_is_ready(flash_dev)) {
+		LOG_ERR("Device %s is not ready.", flash_dev->name);
+		atomic_inc(&completed_threads);
```

Comment:
Not needed here? missing increment will allow to detect that something has failed

Reply:
failure will be detected because of missing LOG_INF("Flash thread has completed");